bindings/haskell/haskell-src/OpenDAL.hs

```diff
@@ -63,6 +67,28 @@ data ErrorCode
 data OpenDALError = OpenDALError {errorCode :: ErrorCode, message :: String}
   deriving (Eq, Show)
 
+data EntryMode = File | Dir | Unknown deriving (Eq, Show)
+
+data BytesContentRange = BytesContentRange
+  { bcrStart :: Integer
+  , bcrEnd :: Integer
+  , bcrTotal :: Integer
+  }
+  deriving (Eq, Show)
+
+data Metadata = Metadata
+  { mMode :: EntryMode
+  , mCacheControl :: Maybe String
+  , mContentDisposition :: Maybe String
+  , mContentLength :: Integer
+  , mContentMD5 :: Maybe String
+  , mContentRange :: Maybe BytesContentRange
```

Comment:
`BytesContentRange` should not be exposed AS-IS. It's not part of OpenDAL's public API. Please use string.

Reply:
Maybe it's better just to ignore it (for now).

Reply:
I have removed `BytesContentRange` in new commit.